agent/worker.go

```diff
@@ -39,24 +42,26 @@ type statusReporterKey struct {
 }
 
 type worker struct {
-	db        *bolt.DB
-	executor  exec.Executor
-	publisher exec.LogPublisher
-	listeners map[*statusReporterKey]struct{}
-	secrets   *secrets
+	db                *bolt.DB
+	taskevents        *events.Broadcaster
```

Comment:
I've updated the LogBroker - however, there's no way to terminate a Worker so it should never leak